lib/rails/auth/controller_methods.rb

```diff
@@ -0,0 +1,20 @@
+require "active_support/hash_with_indifferent_access"
+
+module Rails
+  module Auth
+    # Convenience methods designed to be included in an ActionController::Base subclass
+    # Recommended use: include in ApplicationController
+    module ControllerMethods
+      # Obtain credentials for the current request
+      #
+      # @return [HashWithIndifferentAccess] credentials extracted from the environment
+      #
+      def credentials
```

Comment:
The intent of `credentials` is to expose multiple types of credentials extracted via AuthN middleware, e.g.

```
credentials[:x509]
```

...is the client's (authenticated) X.509 certificate, however the same API could be used to obtain authenticated bearer credentials like cookies or OAuth tokens.